File: 0055-jump-game/0055-jump-game.go
func canJump(nums []int) bool {
    reachable := 0
	for i := 0; i < len(nums); i++ {
		if reachable >= len(nums)-1 {
			return true
		}
		if i > reachable {
			return false
		}
		reachable = max(reachable, nums[i]+i)
	}
	return true
}

File: leetcode/0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: list[int]) -> bool:
        reachable = 0
        for i in range(len(nums)):
            if reachable >= len(nums) - 1:
                return True
                
            if i > reachable:
                return False
            reachable = max(reachable, i + nums[i])
        return True